docs/advanced/extensibleenums.md

```diff
@@ -0,0 +1,220 @@
+# Extensible Enums
+
+Extensible Enums are an enhancement of specific Vanilla enums to allow new entries to be added. This is done by modifying the compiled bytecode of the enum at runtime to add the elements.
+
+## `IExtensibleEnum`
+
+All enums that can have new entries implement the `IExtensibleEnum` interface. This interface acts as a marker to allow the `RuntimeEnumExtender` launch plugin service to know what enums should be transformed.
+
+:::warning
+You should **not** be implementing this interface on your own enums. Use maps or registries instead depending on your usecase.
+:::
+
+### Creating an Enum Entry
+
+To create a new enum entry, the static `create` method should be called. This static method is added to all extensible enums and is transformed at runtime to allow new enums to be added.
+
+:::note
+The `create` method should be called within a static final field if you need to refer to the value or within the main mod constructor if you do not.
+:::
+
+The `create` method contains a name parameter followed by the parameters of a supporting constructor. The name parameter represents the name of the enum constant, which is returned by `#name`. The name of the enum constant should be prefixed with your mod id followed by an underscore (`_`) to avoid conflicts between mods that decide to add the same enum constant (e.g., adding a enum constant called `test` with mod id `examplemod` should be `EXAMPLEMOD_TEST`).
+
+:::note
+Since extensible enums are added at runtime, technically, any UTF-8 character can be used for the name paramter. However, it is recommended to only use valid Java identifiers.
```

Comment:
```suggestion
Since extensible enums are added at runtime, technically, any UTF-8 character can be used for the name parameter. However, it is recommended to only use valid Java identifiers.
```